scripts/generate-functionmap.php

```diff
@@ -0,0 +1,139 @@
+                if ($method->getDeclaringClass() != $reflectionClass && $method->getName() != '__toString') {
+                    continue;
+                }
+
+                if (in_array($method->getName(), $skippedMethods, true)) {
+                    continue;
+                }
+
+                $methodKey = $reflectionClass->getName() . '::' . $method->getName();
+                $functionMap[$methodKey] = $this->getFunctionMapEntry($method);
+            }
+        }
+
+        return $functionMap;
+    }
+
+    private function writeFunctionMap(string $filename, array $functionMap): void
+    {
+        $lines = [];
+        foreach ($functionMap as $methodName => $typeInfo) {
+            $generatedTypeInfo = implode(
+                ', ',
+                array_map(
+                    function (string|int $key, string $value): string {
+                        if (is_int($key)) {
+                            return $this->removeDoubleBackslash(var_export($value, true));
+                        }
+
+                        return sprintf('%s => %s', var_export($key, true), $this->removeDoubleBackslash(var_export($value, true)));
+                    },
+                    array_keys($typeInfo),
+                    array_values($typeInfo)
+                )
+            );
+
+            $lines[] = sprintf(
+                '    %s => [%s],',
+                $this->removeDoubleBackslash(var_export($methodName, true)),
+                $generatedTypeInfo
+            );
+        }
+
+        $fileTemplate = <<<'PHP'
+<?php
+
+$mongoDBFunctionMap = [
+%s
+];
+
+PHP;
+
+        file_put_contents($filename, sprintf($fileTemplate, implode("\n", $lines)));
+    }
+
+    private function filterItems(string $name): bool {
+        $namespaces = ['MongoDB\BSON\\', 'MongoDB\Driver\\'];
+
+        $name = strtolower($name);
+
+        foreach ($namespaces as $namespace) {
+            // Always compare lowercase names, as get_defined_functions lowercases function names by default
+            if (str_starts_with($name, strtolower($namespace))) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private function getFunctionMapEntry(ReflectionFunctionAbstract $function): array {
+        $returnType = match(true) {
+            $function->hasReturnType() => (string) $function->getReturnType(),
+            $function->hasTentativeReturnType() => (string) $function->getTentativeReturnType(),
+            default => 'void',
+        };
+
+        $functionMapEntry = [$returnType];
+
+        foreach ($function->getParameters() as $parameter) {
+            $parameterKey = $parameter->getName();
+            if ($parameter->isOptional()) {
+                $parameterKey .= '=';
+            }
+
+            $parameterType = (string) $parameter->getType();
+            if ($function->getName() === 'unserialize' && $parameter->getName() === 'serialized') {
```

Comment:
Noted that this is sticking around. If you intend to add a conditional declaration in a separate PHPC ticket we can leave this as-is and then come back and remove this later.

Reply:
This has been removed now that we have conditional declarations for the method in question.